app/Http/Controllers/PostController.php

```diff
@@ -2,54 +2,86 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Post;
+use App\Models\Group;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\Support\Facades\Validator;
 
 class PostController extends Controller
 {
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function index()
+    public function __construct()
     {
-        //
+        $this->middleware('auth', ['only' => 'create', 'store', 'destroy']);
     }
 
     /**
-     * Show the form for creating a new resource.
+     * Show the form for creating a new post.
      *
      * @return \Illuminate\Http\Response
      */
-    public function create()
+    public function create($group)
     {
-        //
+        return view('posts.create-post', ['group' => $group]);
```

Comment:
Let's actually pass the whole group to the view so we can access both id for the POST request and the name for display.

```suggestion
        Validator::validate(['id' => $id]), [
            'id' => 'required|int|exists:groups'
        ]);
        
        $group = Group::where('id', $id)->first();
        return view('posts.create-post', ['group' => $group]);
```

Reply:
Solved.